jujugui/static/gui/src/app/components/inspector/inspector.js

```diff
@@ -367,7 +367,10 @@ class Inspector extends React.Component {
           activeComponent={this.state.activeComponent}
           type={this.state.activeChild.headerType}
           title={this.state.activeChild.title}
-          icon={this.state.activeChild.icon} />
+          entityId={this.props.service.get('charm')}
+          icon={this.state.activeChild.icon}
+          changeState={this.props.appState.changeState.bind(this.props.appState)}
```

Comment:
sinon.stub()

Reply:
This isn't a test?

Reply:
changeState seems to be always provided, which contradict teh comment above about it being provided only if a get started is present. As suggested, we could make it required, also provide an additional hasGetStarted boolean, and pass an already parsed URL. I believe that `this.props.service.get('charm')` is a legacy URL, so no need for the try/catch block, let's try with just using fromLegacyString.